### ISSUE TITLE
aarch64: Libreoffice template compilation issue

### ISSUE THREAD

**Issue by BikyAlex** [void-linux/void-packages#35352]
### System
Void Linux glibc aarch64 - Raspberry Pi 4

* xuname:  
  Void 5.10.52_1 aarch64 Unknown uptodate rF
* package:  
  *affected package(s) including the version*: ``LibreOffice 7.2.3.2 template from void-packages/srcpkgs ``

### Expected behavior
  compile successfully

### Actual behavior
  fail to compile

### Steps to reproduce the behavior
```
  cd void-packages
  git pull --rebase upstream master
  ./xbps-src binary-bootstrap
  ./xbps-src pkg libreoffice  
```

### Output:
```
[build RDB] services
[build PKG] svx_gengal
[build CXX] svx/source/gengal/gengal.cxx
[build LNK] Executable/gengal.bin
[build GAL] backgrounds
terminate called after throwing an instance of 'com::sun::star::uno::RuntimeException'
make[1]: *** [/builddir/libreoffice-7.2.3.2/solenv/gbuild/Gallery.mk:56: /builddir/libreoffice-7.2.3.2/workdir/Gallery/backgrounds.done] Error 1
make: *** [Makefile:287: build] Error 2
=> ERROR: libreoffice-7.2.3.2_2: do_build: 'make ${makejobs} ${make_build_target}' exited with 2
=> ERROR:   in do_build() at srcpkgs/libreoffice/template:523
```

---

My question would be, where is "${makejobs}" defined in the template, because it only appears twice and in neither cases is it declared, unless xbps-src has this as an env variable somewhere. "${make_build_target}" is defined though. 

I did clean the hostdir/builddir folder once after the first failure, but that didn't seem to help. 

As a background, if that interests anyone, I'm using the Raspberry Pi 4 as my main PC (been my main for more than 8 months now), but libreoffice is not present in the aarch64 repo. I am assuming that's because libreoffice doesn't work with cross-compilation, so I have to compile it myself natively. I don't mind that, but I can't decode the xbps-src template myself and fix it. Does it work to compile on x86? I don't have an x86 system to test the template there. And I also cannot decode how buildbot works, so I can check the status of LO.

**Comment by ericonr** [comment]
`$makejobs` is defined by xbps-src before sourcing the template.

Is that the full error output? Getting to see the whole thing might help. Can you monitor memory usage during this point as well? I wouldn't be surprised if it's running out of memory during the build.

I haven't built it recently, but yes, it worked on x86. You can check package status in repos in https://voidlinux.org/packages

**Comment by BikyAlex** [comment]
```
/xbps-src -Q pkg libreoffice
=> xbps-src: updating repositories for host (aarch64)...
[*] Updating repository `https://alpha.de.repo.voidlinux.org/current/aarch64/aarch64-repodata' ...
aarch64-repodata: 1586KB [avg rate: 843KB/s]
[*] Updating repository `https://alpha.de.repo.voidlinux.org/current/aarch64/nonfree/aarch64-repodata' ...
[*] Updating repository `https://alpha.de.repo.voidlinux.org/current/aarch64/debug/aarch64-repodata' ...
aarch64-repodata: 711KB [avg rate: 488KB/s]
=> xbps-src: updating software in / masterdir...
=> xbps-src: cleaning up / masterdir...
=> libreoffice-7.2.3.2_2: removing autodeps, please wait...
=> libreoffice-7.2.3.2_2: the following build options are set:
   java: Enable Java support (OFF)
=> libreoffice-7.2.3.2_2: building [meta] for aarch64...
   [host] automake-1.16.5_1: found (https://alpha.de.repo.voidlinux.org/current/aarch64)
   [host] flex-2.6.4_3: found (https://alpha.de.repo.voidlinux.org/current/aarch64)
   [host] git-2.35.0_1: found (https://alpha.de.repo.voidlinux.org/current/aarch64)
   [host] gperf-3.1_3: found (https://alpha.de.repo.voidlinux.org/current/aarch64)
[truncated'
   [host] perl-Archive-Zip-1.68_2: found (https://alpha.de.repo.voidlinux.org/current/aarch64)
   [host] python3-setuptools-60.5.0_1: not found
   [host] fontforge-20201107_2: found (https://alpha.de.repo.voidlinux.org/current/aarch64)
   [host] python3-lxml-4.6.4_2: not found
   [target] CoinMP-devel-1.8.4_1: found (https://alpha.de.repo.voidlinux.org/current/aarch64)
   [target] apr-devel-1.7.0_4: found (https://alpha.de.repo.voidlinux.org/current/aarch64)
   [target] avahi-libs-devel-0.8_2: found (https://alpha.de.repo.voidlinux.org/current/aarch64)
   [target] clucene-devel-2.3.3.4_9: found (https://alpha.de.repo.voidlinux.org/current/aarch64)
   [target] frameworkintegration-devel-5.89.0_1: found (https://alpha.de.repo.voidlinux.org/current/aarch64)
   [target] glyphy-devel-0.0.20190307_2: found (https://alpha.de.repo.voidlinux.org/current/aarch64)
   [target] gpgmepp-devel-1.15.1_2: found (https://alpha.de.repo.voidlinux.org/current/aarch64)
   [target] gst-plugins-base1-devel-1.18.5_1: found (https://alpha.de.repo.voidlinux.org/current/aarch64)
   [target] gtk+-devel-2.24.32_4: found (https://alpha.de.repo.voidlinux.org/current/aarch64)
   [target] gtk+3-devel-3.24.31_1: found (https://alpha.de.repo.voidlinux.org/current/aarch64)
   [target] hunspell-devel-1.7.0_3: found (https://alpha.de.repo.voidlinux.org/current/aarch64)
   [target] hyphen-devel-2.8.8_3: found (https://alpha.de.repo.voidlinux.org/current/aarch64)
   [target] libXt-devel-1.2.1_1: found (https://alpha.de.repo.voidlinux.org/current/aarch64)
   [target] libabw-devel-0.1.3_1: found (https://alpha.de.repo.voidlinux.org/current/aarch64)
   [target] libatomic_ops-devel-7.6.12_1: found (https://alpha.de.repo.voidlinux.org/current/aarch64)
   [target] libbluetooth-devel-5.63_1: found (https://alpha.de.repo.voidlinux.org/current/aarch64)
   [target] libcdr-devel-0.1.7_3: found (https://alpha.de.repo.voidlinux.org/current/aarch64)
[truncated]
   [runtime] libreoffice-common-7.2.3.2_2: not found (subpkg, ignored)
   [runtime] libreoffice-base-7.2.3.2_2: not found (subpkg, ignored)
   [runtime] libreoffice-calc-7.2.3.2_2: not found (subpkg, ignored)
   [runtime] libreoffice-draw-7.2.3.2_2: not found (subpkg, ignored)
   [runtime] libreoffice-fonts-7.2.3.2_2: not found (subpkg, ignored)
   [runtime] libreoffice-gnome-7.2.3.2_2: not found (subpkg, ignored)[
[truncated]   [runtime] libreoffice-writer-7.2.3.2_2: not found (subpkg, ignored)
   [runtime] hunspell-1.7.0_3: found (https://alpha.de.repo.voidlinux.org/current/aarch64)
   [runtime] hyphen-2.8.8_3: found (https://alpha.de.repo.voidlinux.org/current/aarch64)
   [runtime] mythes-1.2.4_5: found (https://alpha.de.repo.voidlinux.org/current/aarch64)
   [runtime] openldap-2.4.58_1: found (https://alpha.de.repo.voidlinux.org/current/aarch64)
   [runtime] libreoffice-i18n-en-US-7.2.3.2_2: not found (subpkg, ignored)
[truncated]
=> libreoffice-7.2.3.2_2: installing host dependencies: automake-1.16.5_1 flex-2.6.4_3 git-2.35.0_1 gperf-3.1_3 hyphen-2.8.8_3 icu-70.1_1 libtool-2.4.6_6 open
ldap-2.4.58_1 which-2.21_3 gettext-0.21_4 xz-5.2.5_2 perl-Archive-Zip-1.68_2 pkg-config-0.29.2_3 qt5-qmake-5.15.3+20211001_2 sane-1.0.32_1 ucpp-1.3.2_2 unzip-
6.0_14 wget-1.21.2_1 zip-3.0_6 fontforge-20201107_2 python3-setuptools-60.5.0_1 python3-lxml-4.6.4_2 ...
=> libreoffice-7.2.3.2_2: installing target dependencies: CoinMP-devel-1.8.4_1 apr-devel-1.7.0_4 avahi-libs-devel-0.8_2 clucene-devel-2.3.3.4_9 frameworkinteg
ration-devel-5.89.0_1 glyphy-devel-0.0.20190307_2 gpgmepp-devel-1.15.1_2 gst-plugins-base1-devel-1.18.5_1 gtk+-devel-2.24.32_4 gtk+3-devel-3.24.31_1 hunspell-
devel-1.7.0_3 hyphen-devel-2.8.8_3 libXt-devel-1.2.1_1 libabw-devel-0.1.3_1 libatomic_ops-devel-7.6.12_1 libbluetooth-devel-5.63_1 libcdr-devel-0.1.7_3 libe-b
ook-devel-0.1.3_10 libepubgen-devel-0.1.1_2 libetonyek-devel-0.1.9_4 libexttextcat-devel-3.4.5_2 libfbclient3-devel-3.0.6.33328_1 libfreehand-devel-0.1.2_3 li
bgcrypt-devel-1.9.4_1 libgltf-devel-0.1.0_2 libldap-devel-2.4.58_1 libmariadbclient-devel-10.5.10_2 libmspub-devel-0.1.4_8 libmwaw-devel-0.3.16_2 libnumbertex
t-devel-1.0.7_1 libodfgen-devel-0.1.8_1 liborcus-devel-0.16.1_2 libpagemaker-devel-0.0.4_2 libqxp-devel-0.0.2_6 librsvg-devel-2.52.2_1 libvisio-devel-0.1.7_4
libwpg-devel-0.3.3_1 libwps-devel-0.4.12_1 lpsolve-devel-5.5.2.5_2 mdds-1.5.0_3 mythes-devel-1.2.4_5 neon-devel-0.31.2_2 nss-devel-3.74_1 postgresql-libs-deve
l-14.1_1 qt5-devel-5.15.3+20211001_2 redland-devel-1.0.17_6 sane-devel-1.0.32_1 telepathy-glib-devel-0.24.2_1 unixodbc-devel-2.3.9_1 vigra-devel-1.11.1_6 libc
ppunit-devel-1.15.1_1 qrcodegen-devel-1.6.0_4 xmlsec1-devel-1.2.33_1 poppler-cpp-devel-21.07.0_1 libxml2-devel-2.9.10_6 python3-devel-3.10.2_1 ...
=> libreoffice-7.2.3.2_2: running pre-build hook: 02-script-wrapper ...
=> libreoffice-7.2.3.2_2: running do_build ...
mkdir -p /builddir/libreoffice-7.2.3.2/instdir
/builddir/libreoffice-7.2.3.2/solenv/bin/install-gdb-printers -a /builddir/libreoffice-7.2.3.2/instdir -c
make -j 1  -rs -f /builddir/libreoffice-7.2.3.2/Makefile.gbuild build

[build PRL] CustomTarget/postprocess/images/sorted.lst
[build PRL] CustomTarget/postprocess/images/commandimagelist.ilst[build PRL] CustomTarget/postprocess/images/images_sukapura.zip
[build PRL] CustomTarget/postprocess/images/images_sukapura_svg.zip
[build GAL] backgrounds
terminate called after throwing an instance of 'com::sun::star::uno::RuntimeException'
make[1]: *** [/builddir/libreoffice-7.2.3.2/solenv/gbuild/Gallery.mk:56: /builddir/libreoffice-7.2.3.2/workdir/Gallery/backgrounds.done] Error 1
make: *** [Makefile:287: build] Error 2
=> ERROR: libreoffice-7.2.3.2_2: do_build: 'make ${makejobs} ${make_build_target}' exited with 2
=> ERROR:   in do_build() at srcpkgs/libreoffice/template:523
```

This is basically the whole output. The compilation stops at building backgrounds. This is the output of the 4th retry, where this time I used -Q, as described in the void-packages Manual.md.

I will monitor the memory usage. I have the 8GB of RAM RPi model, but Firefox and Thunderbird to tend to monopolize the RAM. They don't go over 4GB, but they cache a lot, up to the point of going into swap (the swap was like 56k usage, out of 8GB).

I'll try rebuilding with them closed, but I somewhat doubt it's going to work.

Thank you for your help, I appreciate it. Hopefully we can figure out why it doesn't compile on aarch64 (one of the previous 7.1 released did compile before, but I had to remove it, because it was too old and I had lib conflicts).

**Comment by BikyAlex** [comment]
Update. So I had tried compiling LO again. I remembered when I was (cross-)compiling Void Linux for the Pi 4 on my old x86 PC when I still had one, I had issues with my long uptimes and probably with unapplied updates due to no reboots. This time I had 9 days uptime on my Pi. A reboot solved that before.

So a reboot later and trying to recompile... and it fails with the same error.
```
=> libreoffice-7.2.3.2_2: running pre-build hook: 02-script-wrapper ...       
=> libreoffice-7.2.3.2_2: running do_build ...
mkdir -p /builddir/libreoffice-7.2.3.2/instdir
/builddir/libreoffice-7.2.3.2/solenv/bin/install-gdb-printers -a /builddir/libreoffice-7.2.3.2/instdir -c
make -j 1  -rs -f /builddir/libreoffice-7.2.3.2/Makefile.gbuild build
[build PRL] CustomTarget/postprocess/images/sorted.lst
[build PRL] CustomTarget/postprocess/images/commandimagelist.ilst
[build PRL] CustomTarget/postprocess/images/images_breeze.zip
[build PRL] CustomTarget/postprocess/images/images_breeze_dark.zip

[truncated]

[build PRL] CustomTarget/postprocess/images/images_sifr_svg.zip
[build PRL] CustomTarget/postprocess/images/images_sukapura.zip
[build PRL] CustomTarget/postprocess/images/images_sukapura_svg.zip
[build GAL] backgrounds
terminate called after throwing an instance of 'com::sun::star::uno::RuntimeException'
make[1]: *** [/builddir/libreoffice-7.2.3.2/solenv/gbuild/Gallery.mk:56: /builddir/libreoffice-7.2.3.2/workdir/Gallery/backgrounds.done] Error 1
make: *** [Makefile:287: build] Error 2
=> ERROR: libreoffice-7.2.3.2_2: do_build: 'make ${makejobs} ${make_build_target}' exited with 2
=> ERROR:   in do_build() at srcpkgs/libreoffice/template:523
```

I had nothing else open, other than my DE and a terminal, so I was using around 150 MB at best, then left LibreOffice to compile and I haven't touched the PC. It should have had around 7.5 GB dedicated to compiling. It's likely that the gallery stuff just doesn't like aarch64.

Any ideas? Should try compiling in a chroot? Or should I try compiling in a Void aarch64 LXD container? I still slightly doubt it's going to make a difference. I don't really want to expend development resources from Void on this, which is why I added low priority in the title. If I knew what to fix, I'd fix it and contribute back.

One thing I researched is that the com:sun:star:uno exception is a Java thing. Maybe some Java dependencies are made for x86 and they fail to run on ARM. That could be one thing. Another one I found is that some people had issues opening up libreoffice because their ~/.config/ folder was for some reason not owned by them. And the last one I found was on OpenOffice that someone was using the UnoRuntime API in a wrong way.

I guess I may as well give LO compilation a try in an LXD clean environment, so I can make sure it's not something from the xbps-src side, related to one of the previous LO compilations.

**Comment by BikyAlex** [comment]
This time I just moved to an lxd container, git cloned void-packages, binary-bootstrapped and compiled LO again, with the same error.
```
[build GAL] backgrounds
terminate called after throwing an instance of 'com::sun::star::uno::RuntimeException'
make[1]: *** [/builddir/libreoffice-7.2.3.2/solenv/gbuild/Gallery.mk:56: /builddir/libreoffice-7.2.3.2/workdir/Gallery/backgrounds.done] Error 1
make: *** [Makefile:287: build] Error 2
=> ERROR: libreoffice-7.2.3.2_2: do_build: 'make ${makejobs} ${make_build_target}' exited with 2
=> ERROR:   in do_build() at srcpkgs/libreoffice/template:523
```
It did take about 24h to get to this point though, because it actually compiled a lot of stuff, but in the end, it was no use.

I guess I will have to wait for the next release and template, hoping that it will compile for aarch64. And unfortunately for me, I make use of LO automatic color row, which Calligra lacks, so using this (which is available in the repo) is out of the question. 

Was there any work done on this, or should I close this issue?

**Comment by ericonr** [comment]
Sorry for the lack of answers here.

I think you should leave the issue open, though I believe I remember people who successfully built LibreOffice on aarch64... Given the error, you could attempt building with `-o ~java` to see if it completes when built without java.

To clean up old build traces, removing the `masterdir/` directory and binary bootstrapping again should be the most you need to do

**Comment by BikyAlex** [comment]
I haven't used LO in a while, I wish I could though. I have cleaned the masterdir previously and even just created a new folder and git cloned this repo, but no dice. 

I didn't know I can build it without java, I frankly don't even need java from what I know, so I'll try. Reading the help page on xbps-src, I didn't notice the -o option. I'll grab the latest LO and come back in a day after it finishes compiling.

**Comment by BikyAlex** [comment]
Reporting in. Updated the repo, deleted the masterdir, binary-bootstrapped, compiled LO 7.3.3.2 (newer version, obviously) without Java. Success. TYVM Eric, you're a genius (or I'm a brainlet, I should have thought of this and read the xbps-src --help more carefully).

LO Calc works, which is all I need. It's technically case closed for me, but I don't know if we should keep it open in order to fix the template file, or if we should close it and just hope people who want LO on their ARM computers will stumble across this thread and compile it without Java.

**Comment by ericonr** [comment]
it's Érico ;)

I'm glad it worked! We could consider disabling  Java by default for the aarch64 target, pending on more investigation... I'm not sure what the best path forward would be. Thank you for the quick answers and testing :)

And no worries, thinking about build options is not always obvious. I was quite familiar with both xbps-src and this template already :)

**Comment by classabbyamp** [comment]
is this still an issue?

**Comment by BikyAlex** [comment]
Last I tried compiling LO on aarch64, it was still failing. Let me give it another go, I will reply back.

Although by this point, the number of people who use LO on aarch64 can probably be counted on a hand. I have been contemplating of just buying a small powered x86 box for my needs, as my trusty RPi 4 has been feeling sluggish lately in other tasks too.

I am not sure how worth it it is at this point to maintain the arm version of Libre Office, if it works, I will report it, otherwise, I will keep you posted, and we can probably close it as unresolved.

**Comment by BikyAlex** [comment]
Ok, just xbps zapped, binary-bootstrapped, then built LO. Failed pretty fast in the process.

```
CMake Warning at /usr/share/ECM/modules/ECMQueryQt.cmake:97 (message):
  Failed call: "QT_INSTALL_PREFIX"
Call Stack (most recent call first):
  /usr/share/ECM/kde-modules/KDEInstallDirs5.cmake:244 (ecm_query_qt)
  /usr/share/ECM/kde-modules/KDEInstallDirs.cmake:15 (include)
  CMakeLists.txt:15 (include)


CMake Error at /usr/share/ECM/modules/ECMQueryQt.cmake:98 (message):
  Qt5 qmake call failed: 101
Call Stack (most recent call first):
  /usr/share/ECM/kde-modules/KDEInstallDirs5.cmake:244 (ecm_query_qt)
  /usr/share/ECM/kde-modules/KDEInstallDirs.cmake:15 (include)
  CMakeLists.txt:15 (include)


-- Configuring incomplete, errors occurred!
See also "/builddir/kpackage-5.101.0/build/CMakeFiles/CMakeOutput.log".
=> ERROR: kpackage-5.101.0_1: do_configure: 'CFLAGS="-DNDEBUG ${CFLAGS/ -pipe / }" CXXFLAGS="-DNDEBUG ${CXXFLAGS/ -pipe / }" cmake ${cma
ke_args} ${configure_args} ${LIBS:+-DCMAKE_C_STANDARD_LIBRARIES="$LIBS"} ${LIBS:+-DCMAKE_CXX_STANDARD_LIBRARIES="$LIBS"} ${wrksrc}/${bui
ld_wrksrc}' exited with 1
=> ERROR:   in do_configure() at common/build-style/cmake.sh:75
```








Content of the masterdir/builddir/kpackage-5.101.0/build/CMakeFiles/CMakeOutput.log file:
https://gist.github.com/BikyAlex/6d13ace6734238fa69c5281afaf9c769


cmake.sh line 75 is:

```
        # Remove -pipe: https://gitlab.kitware.com/cmake/cmake/issues/19590
        CFLAGS="-DNDEBUG ${CFLAGS/ -pipe / }" CXXFLAGS="-DNDEBUG ${CXXFLAGS/ -pipe / }" \
                cmake ${cmake_args} ${configure_args} \
                ${LIBS:+-DCMAKE_C_STANDARD_LIBRARIES="$LIBS"} \
                ${LIBS:+-DCMAKE_CXX_STANDARD_LIBRARIES="$LIBS"} \
                ${wrksrc}/${build_wrksrc}
```

I think we should probably close this. This is not even a LibreOffice compilation issue now, we are getting into kpackage territory. I honestly feel like it is more of a hassle for the maintainers, I do not feel like bothering you with having to maintain this for 1 or 5 users. I wish I could help, but I do not have a lot of free time to contribute, or even to fix my own stuff, I have mostly been ignoring this compilation and tried working around it.

**Comment by classabbyamp** [comment]
i think that kpackage issue is probably resolved by now, it was probably just building the new version while you were testing

**Comment by BikyAlex** [comment]
```
configure: VCLplugs to be built: gen gtk3 gtk3_kde5 gtk4 kf5 qt5 qt6
checking for gtk+-3.0 >= 3.20 gtk+-unix-print-3.0 gmodule-no-export-2.0 glib-2.0 >= 2.38 atk >= 2.28.1 cairo... yes
checking for gtk4 gmodule-no-export-2.0 glib-2.0 >= 2.38 cairo atk... yes
checking for gobject-introspection... configure: error: gobject-introspection-1.0 is not installed
=> ERROR: libreoffice-7.4.2.3_2: do_configure: './configure ${opts} --with-lang="${_languages}"' exited with 1
=> ERROR:   in do_configure() at srcpkgs/libreoffice/template:504
```

Line 504:
`./configure ${opts} --with-lang="${_languages}"`
This is the point where it was failing me last time too. I tried commenting all the languages besides English, which is what I need anyway, but it did not work, it fails to configure. I did not think it may have been the opts though, let me try and fine what opts it was using.

**Comment by BikyAlex** [comment]
```
=> libreoffice-7.4.2.3_2: running pre-configure hook: 02-script-wrapper ...
=> libreoffice-7.4.2.3_2: running do_configure ...
Skipping configure process../configure \
--prefix=/usr \
--sysconfdir=/etc \
--sbindir=/usr/bin \
--bindir=/usr/bin \
--mandir=/usr/share/man \
--infodir=/usr/share/info \
--localstatedir=/var \
--with-vendor='VoidLinux' \
--with-help --disable-odk \
--disable-online-update \
--enable-python=system \
--enable-gtk3 \
--enable-introspection \
--enable-gtk4 \
--enable-qt5 \
--enable-qt6 \
--enable-kf5 \
--enable-gtk3-kde5 \
--disable-dependency-tracking \
--enable-release-build \
--enable-split-app-modules \
--disable-dconf \
--with-system-libs \
--with-system-headers \
--with-fonts=yes \
--without-java \
--with-tls=nss \
--without-system-abseil \
--without-system-box2d \
--without-system-libcmis \
--without-system-libzmf \
--without-system-libstaroffice \
--without-system-libtommath \
--without-system-jfreereport \
--without-system-beanshell \
--without-system-dragonbox \
--without-system-libfixmath \
--with-myspell-dicts \
--enable-build-opensymbol \
--with-system-ucpp=yes \
--with-external-dict-dir=/usr/share/hunspell \
--with-external-hyph-dir=/usr/share/hyphen \
--with-external-thes-dir=/usr/share/mythes \
--with-parallelism=4 \
--with-lang= af am ar as ast be bg bn bn-IN bo br brx bs ca ca-valencia cs cy da de dgo dz el en-GB en-US en-ZA eo es et eu fa fi fr ga gd gl gu gug he hi hr hu id is it ja ka kk km kmr-Latn kn ko kok ks lb lo lt lv mai mk ml mn mni mr my nb ne nl nn nr nso oc om or pa-IN pl pt pt-BR ro ru rw sa-IN sat sd si sid sk sl sq sr sr-Latn ss st sv sw-TZ ta te tg th tn tr ts tt ug uk uz ve vi xh zh-CN zh-TW zu
********************************************************************
*
*   Running LibreOffice build configuration.
*
********************************************************************
```

It took a bit, because I wanted to make it a bit more readable, I put each option on a new line. I cannot figure out where it is failing. 

I think might have something to do with this:

masterdir/builddir/libreoffice-7.4.2.3/config.log

https://gist.github.com/BikyAlex/9feb01e9ec6c1429a36f2ccf5a0376e2

```
configure:8306: cc -c -fstack-clash-protection -D_FORTIFY_SOURCE=2 -march=armv8-a -O2 -pipe    -fdebug-prefix-map=/builddir/libreoffice-
7.4.2.3=.     conftest.c >&5
conftest.c:29:16: error: unknown type name 'not'
   29 |                not a universal capable compiler
      |                ^~~
conftest.c:29:22: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'universal'
   29 |                not a universal capable compiler
      |                      ^~~~~~~~~
conftest.c:29:22: error: unknown type name 'universal'
configure:8306: $? = 1
```

**Comment by BikyAlex** [comment]
An xbp-src zap, git pull, binary-bootstrap and pkg -o -java LO later, the build still fails, as expected.

```
checking for gtk+-3.0 >= 3.20 gtk+-unix-print-3.0 gmodule-no-export-2.0 glib-2.0 >= 2.38 atk >= 2.28.1 cairo... yes
checking for gtk4 gmodule-no-export-2.0 glib-2.0 >= 2.38 cairo atk... yes
checking for gobject-introspection... configure: error: gobject-introspection-1.0 is not installed
=> ERROR: libreoffice-7.4.3.2_1: do_configure: './configure ${opts} --with-lang="${_languages}"' exited with 1
=> ERROR:   in do_configure() at srcpkgs/libreoffice/template:506
```

Nothing jumped to my eye in masterdir/builddir/libreoffice-7.4.3.2/config.log. It just exists "configure: exit 1" without anything specified.

I'm closing this as unresolved. I'm moving back to x86. I'm probably the only one using LO on aarch64 anyway. I'll still use Void on my arm servers, it's rock solid there. Well, Void in general is rock solid, I'd like to thank everyone for all their efforts on the distro and the package maintainers. This LO thing is probably not worth the time investment.
